openedx/core/process_warnings.py

```diff
@@ -0,0 +1,241 @@
+from djangolib.markup import HTML, Text  # noqa pylint: disable=import-error
+
+columns = [
+    "message",
+    "category",
+    "filename",
+    "lineno",
+    "high_location",
+    "label",
+    "num",
+    "deprecated",
+]
+columns_index_dict = {key: index for index, key in enumerate(columns)}
+
+
+pp = pprint.PrettyPrinter(indent=4, depth=2)
+
+
+def seperate_warnings_by_location(warnings_data):
+    """
+    Warnings originate from multiple locations, this function takes in list of warning objects
+    and seperates them based on their filename location
+    """
+
+    # first create regex for each know file location
+    warnings_locations = {
+        ".*/python\d\.\d/site-packages/.*\.py": "python",  # noqa pylint: disable=W1401
+        ".*/edx-platform/lms/.*\.py": "lms",  # noqa pylint: disable=W1401
+        ".*/edx-platform/openedx/.*\.py": "openedx",  # noqa pylint: disable=W1401
+        ".*/edx-platform/cms/.*\.py": "cms",  # noqa pylint: disable=W1401
+        ".*/edx-platform/common/.*\.py": "common",  # noqa pylint: disable=W1401
+    }
+
+    # seperate into locations flow:
+    #  * iterate through each wanring_object, see if its filename matches any regex in warning locations.
+    #  * If so, change high_location index on warnings_object to location name
+    for warnings_object in warnings_data:
+        warning_origin_located = False
+        for key in warnings_locations:
+            if (
+                re.search(key, warnings_object[columns_index_dict["filename"]])
+                is not None
+            ):
+                warnings_object[
+                    columns_index_dict["high_location"]
+                ] = warnings_locations[key]
+                warning_origin_located = True
+                break
+        if not warning_origin_located:
+            warnings_object[columns_index_dict["high_location"]] = "other"
+    return warnings_data
+
+
+def convert_warning_dict_to_list(warning_dict):
+    """
+    converts our data dict into our defined list based on columns defined at top of this file
+    """
+    output = []
+    for column in columns:
+        if column in warning_dict:
+            output.append(warning_dict[column])
+        else:
+            output.append(None)
+    output[columns_index_dict["num"]] = 1
+    return output
+
+
+def read_warning_data(dir_path):
+    """
+    During test runs in jenkins, multiple warning json files are output. This function finds all files
+    and aggregates the warnings in to one large list
+    """
+    # pdb.set_trace()
+    dir_path = os.path.expanduser(dir_path)
+    # find all files that exist in given directory
+    files_in_dir = [
+        f for f in os.listdir(dir_path) if os.path.isfile(os.path.join(dir_path, f))
+    ]
+    warnings_files = []
+
+    # TODO(jinder): currently this is hardcoded in, maybe create a constants file with info
+    # THINK(jinder): but creating file for one constant seems overkill
```

Comment:
Maybe just move to a constant at the top of the file?

Reply:
The issue is that there needs to be a way to share file names between pytest_hooks.py(creates the files) and process_warnings.py(processes them).